libs/langgraph/langgraph/func/__init__.py

```diff
@@ -376,37 +476,69 @@ async def agen_wrapper(
             is not inspect.Signature.empty
             else Any
         )
-        output_type = (
-            sig.return_annotation
-            if sig.return_annotation is not inspect.Signature.empty
-            else Any
-        )
+
+        def _pluck_return_value(value: Any) -> Any:
+            """Extract the return_ value the entrypoint.final object or pass through."""
+            return value.value if isinstance(value, entrypoint.final) else value
+
+        def _pluck_save_value(value: Any) -> Any:
+            """Extract the save value from the entrypoint.final object or pass through."""
+            return value.save if isinstance(value, entrypoint.final) else value
+
+        output_type, save_type = Any, Any
+        if sig.return_annotation is not inspect.Signature.empty:
+            # User does not parameterize entrypoint.final properly
+            if (
+                sig.return_annotation is entrypoint.final
```

Comment:
Does signature() resolve string annotations? If not this can also be the string entrypoint.final.

Reply:
Good catch I'll check tomorrow